src/components/Courses/components/SearchBar/SearchBar.jsx

```diff
@@ -0,0 +1,29 @@
+import { Input } from '../../../../common/Input/Input';
+import { Button } from '../../../../common/Button/Button';
+import styles from './SearchBar.module.scss';
+import { useState } from 'react';
+
+export const SearchBar = ({ onSearch }) => {
+	const [search, setSearch] = useState('');
+	const handleSearch = () => {
+		onSearch(search);
+	};
+	const handleInputChange = ({ target: { value } }) => {
+		if (search && !value) {
+			onSearch('');
+		}
+		setSearch(value);
+	};
+	return (
+		<div className={styles.main}>
+			<Input
+				onChange={handleInputChange}
+				name='searchInput'
+				placeholderText='Enter course name...'
+			/>
+			<Button onClick={handleSearch} primary>
```

Comment:
if you use "controlled input" here you will be able to remove one of the handler functions